### PR TITLE
2023/07/10 organising folders according to app architecture

### PR DESCRIPTION
**Pull request by juskek**
For [ticket](https://trello.com/c/y5gBAaUX/113-12-timebox-1hr-x-1-dev-organising-folders-according-to-app-architecture)

### CODE REVIEW

File: src/data/selected-countries/TestSelectedCountriesRepository.ts
Comment (rthandi) [inline src/data/selected-countries/TestSelectedCountriesRepository.ts:1]: Should this have Test in the name? Doesn't seem like it is a spec so the name might be confusing

File: src/background/background.ts
Comment (rthandi) [inline src/background/background.ts:1]: what does background refer to here?

File: src/data/selected-countries/TestSelectedCountriesRepository.ts
Comment (juskek) [reply src/data/selected-countries/TestSelectedCountriesRepository.ts:1]: It's part of the repository pattern (see image below), and so for the interface ISelectedCountriesRepository, there are two implementations:

1. SelectedCountriesRepository (used in dev and production envs, communicates with the external data source)
2. TestSelectedCountriesRepository (used in test env, mocks the behaviour of the external data source when `npm test` is run)

So yes, the Test in the name is intentional :) 

image

File: src/background/background.ts
Comment (juskek) [reply src/background/background.ts:1]: background processes, it's the default name for the background file in chrome extensions - I should probably add a README about this 👍

File: src/utils/helpers/calculateAverageSpecificEmissions.ts
[the remark this answers is not in the document]
Comment (juskek) [reply src/utils/helpers/calculateAverageSpecificEmissions.ts:1]: My first instinct is to usually keep files short and sweet, so I did not group them together.

I do however like the idea of keeping it close to where they are used, will push a new commit!

**Review by rthandi — APPROVED** [review]

**Pull request merged**